### PR TITLE
Guard assert-related functions

### PR DESCRIPTION
**Pull request by bzsolt**
JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/ecma/base/ecma-helpers-string.c [hunk 1]
```diff
@@ -1366,8 +1368,10 @@ ecma_is_string_magic (const ecma_string_t *string_p, /**< ecma-string */
      * should return ecma-string with ECMA_STRING_CONTAINER_MAGIC_STRING
      * container type if new ecma-string's content is equal to one of magic strings.
      */
+#ifndef JERRY_NDEBUG
     JERRY_ASSERT (ecma_string_get_length (string_p) > LIT_MAGIC_STRING_LENGTH_LIMIT
                   || !ecma_is_string_magic_longpath (string_p, out_id_p));
+#endif /* !JERRY_NDEBUG */
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-helpers-string.c]: We need this until the assertion is changed?

Comment (bzsolt) [inline jerry-core/ecma/base/ecma-helpers-string.c]: I would to keep the assertion in the current form, so it isn't a temporary change.

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**